atuin-server/src/handlers/history.rs

```diff
@@ -10,16 +10,18 @@ use tracing::{debug, error, instrument};
 use super::{ErrorResponse, ErrorResponseStatus};
 use crate::{
     calendar::{TimePeriod, TimePeriodInfo},
-    database::{Database, Postgres},
-    models::{NewHistory, User},
+    database::{DatabaseExtension, DatabaseWrapped},
+    models::{NewHistory, User as ModelUser},
 };
 
 use atuin_common::api::*;
 
+type User<T> = DatabaseWrapped<ModelUser, T>;
```

Comment:
What's the point of this exactly? I see no reason the user type needs to be coupled to the DB

Reply:
👋 @conradludgate I agree! I got stuck implementing `FromRequest` for `User` for both Postgres and Sqlite:

https://github.com/ellie/atuin/pull/491/files?diff=unified&w=0#diff-8da1109a66fb471557b348bd52fcbb14936eba4d31a11d3b1fa30b4218be1bf0R28-R31

I couldn't find a way to retrieve the correct type via `extensions` on the request. Maybe boxing the value could work?
There is no real data coupling btw, it is implemented as `PhantomData`.

If you have an idea / suggestion how to fix this I gladly take a stab at it.